File: packages/go-sdk/operations/event_triggers.go
package operations

import "context"

// EventTriggersService provides event trigger management operations.
type EventTriggersService struct{ r Requester }

func NewEventTriggersService(r Requester) *EventTriggersService {
	return &EventTriggersService{r: r}
}

func (s *EventTriggersService) ListEvents(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/events", query, nil, nil, &result)
	return result, err
}

func (s *EventTriggersService) GetEvent(ctx context.Context, eventKey string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", PathParams("/v1/events/{eventKey}", map[string]string{"eventKey": eventKey}), nil, nil, nil, &result)
	return result, err
}

func (s *EventTriggersService) DeleteEvent(ctx context.Context, eventKey string) error {
	return s.r.DoRequest(ctx, "DELETE", PathParams("/v1/events/{eventKey}", map[string]string{"eventKey": eventKey}), nil, nil, nil, nil)
}

func (s *EventTriggersService) SendEvent(ctx context.Context, eventKey string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/events/{eventKey}/send", map[string]string{"eventKey": eventKey}), nil, nil, body, &result)
	return result, err
}

func (s *EventTriggersService) SendPrefix(ctx context.Context, prefix string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/events/prefix/{prefix}/send", map[string]string{"prefix": prefix}), nil, nil, body, &result)
	return result, err
}

func (s *EventTriggersService) PurgeEvent(ctx context.Context, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", "/v1/events/purge", nil, nil, body, &result)
	return result, err
}

func (s *EventTriggersService) GetStat(ctx context.Context) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/events/stats", nil, nil, nil, &result)
	return result, err
}

File: packages/rust-sdk/src/operations/event_triggers.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use crate::http::substitute_path_params;
use std::sync::Arc;

pub struct EventTriggersService {
    client: Arc<StraitClient>,
}

impl EventTriggersService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn list_events(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/event-triggers/events", query, None, None).await
    }

    pub async fn get_event(&self, event_key: &str) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/event-triggers/events/{eventKey}", &[("eventKey", event_key)]);
        self.client.do_request("GET", &path, None, None, None).await
    }

    pub async fn delete_event(&self, event_key: &str) -> Result<(), StraitError> {
        let path = substitute_path_params("/v1/event-triggers/events/{eventKey}", &[("eventKey", event_key)]);
        self.client.do_request_no_content("DELETE", &path, None, None, None).await
    }

    pub async fn send_event(&self, event_key: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/event-triggers/events/{eventKey}/send", &[("eventKey", event_key)]);
        self.client.do_request("POST", &path, None, None, Some(body)).await
    }

    pub async fn send_prefix(&self, prefix: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/event-triggers/prefix/{prefix}/send", &[("prefix", prefix)]);
        self.client.do_request("POST", &path, None, None, Some(body)).await
    }

    pub async fn purge_event(&self, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("POST", "/v1/event-triggers/purge", None, None, Some(body)).await
    }

    pub async fn get_stat(&self) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/event-triggers/stats", None, None, None).await
    }
}
